src/PolymakeWrap.jl

```diff
@@ -96,17 +96,64 @@ include("convert.jl")
 include("sets.jl")
 # to be moved to Vectors/Matrices
 pm_Integer(b::BigInt) = new_pm_Integer(b)
-pm_Rational(num::BigInt, den::BigInt) = pm_Rational(pm_Integer(num), pm_Integer(den))
+pm_Rational(num::T, den::T) where T<:Integer = 
+    pm_Rational(pm_Integer(num), pm_Integer(den))
+pm_Rational(r::R) where R<:Rational = pm_Rational(numerator(r), denominator(r))
+Base.one(i::T) where T <:pm_Integer = pm_Integer(1)
+pm_Rational(int::Integer) = pm_Rational(int, one(int))
+
+Base.promote_rule(::Type{T}, ::Type{Polymake.pm_IntegerAllocated}) where T<:Integer = pm_Integer
+Base.promote_rule(::Type{T}, ::Type{Polymake.pm_RationalAllocated}) where T <: Union{Integer, Rational} = pm_Rational
+
+convert(::Type{pm_Rational}, int::Integer) = pm_Rational(int)
+
+for (pm_T, Abstract_T) in [
+            (:pm_Vector, :AbstractVector),
+            (:pm_Matrix, :AbstractMatrix),
+            ]
+    @eval begin
+        function $(pm_T)(v::$(Abstract_T){T}) where T<:Integer
+            res = $(pm_T){pm_Integer}(size(v)...)
+            res .= v
+            return res
+        end
+
+        function $(pm_T)(v::$(Abstract_T){T}) where T<:Rational
+            res = $(pm_T){pm_Rational}(size(v)...)
+            res .= v
+            return res
+        end
+        
+        $(pm_T){T}(v::$(Abstract_T)) where T = $(pm_T)(convert($(Abstract_T){T}, v))
+    end
+end
 
 Base.size(v::pm_Vector) = (length(v),)
 Base.size(m::pm_Matrix) = (Polymake.rows(m), Polymake.cols(m))
 
 function Base.getindex(V::pmV, n::Integer) where pmV <: pm_Vector 
     1 <= n <= length(V) || throw(BoundsError(V, n))
-    Polymake._getindex(V, n)
+    return Polymake._getindex(V, Int(n))
 end
-function Base.setindex!(V::pmV, val, n::Int) where {T, pmV <: pm_Vector{T}}
+
+function Base.setindex!(V::pmV, val, n::Integer) where {T, pmV <: pm_Vector{T}}
     1 <= n <= length(V) || throw(BoundsError(V, n))
-    return Polymake._setindex!(V, T(val), n)
+    Polymake._setindex!(V, T(val), Int(n))
+    return V
+end
+
+function Base.getindex(M::pmM, i::Integer, j::Integer) where pmM <: pm_Matrix 
+    1 <= i <= Polymake.rows(M) || throw(BoundsError(M, [i,j]))
+    1 <= j <= Polymake.cols(M) || throw(BoundsError(M, [i,j]))
+    return Polymake._getindex(M, Int(i), Int(j))
 end
```

Comment:
The nice way would be to allow other to avoid the bounds checks if they want. This can be accomplished as follows:
```julia
Base.@propagate_inbounds function Base.getindex(M::pm_Matrix , i::Integer, j::Integer)
     @boundscheck 1 <= i <= Polymake.rows(M) || throw(BoundsError(M, [i,j]))
     @boundscheck 1 <= j <= Polymake.cols(M) || throw(BoundsError(M, [i,j]))
     return Polymake._getindex(M, Int(i), Int(j))
 end
```